backend/dataall/modules/s3_datasets/services/dataset_service.py

```diff
@@ -105,20 +105,29 @@ def check_dataset_account(session, environment):
         return True
 
     @staticmethod
-    def check_imported_resources(dataset: S3Dataset):
+    def check_imported_resources(dataset: S3Dataset, data: dict = {}):
+        # check that resource names are valid
+        if dataset.S3BucketName:
+            NamingConventionService(
+                target_uri=dataset.datasetUri,
+                target_label=dataset.S3BucketName,
+                pattern=NamingConventionPattern.S3,
+            ).validate_imported_name()
+
+        if dataset.importedGlueDatabase and dataset.GlueDatabaseName:
```

Comment:
nit: just check `if dataset.importedGlueDatabase:` no need to have `and dataset.GlueDatabaseName` based on previous discussion